cmd/kube-scheduler/app/options/options.go

```diff
@@ -200,7 +201,7 @@ func (o *Options) ApplyTo(c *schedulerappconfig.Config) error {
 		}
 	}
 
-	return nil
+	return utilfeature.DefaultMutableFeatureGate.SetFromMap(o.ComponentConfig.FeatureGates)
```

Comment:
this should be `c.ComponentConfig`

But there is still problems of precedence that need to be figured out. I'll get back to you in a bit.

Reply:
Added validation for DefaultMutableFeatureGate